python/triton/third_party/hip/hip_backend.py

```diff
@@ -274,22 +274,30 @@ def get_amdgcn_bitcode_paths(gfx_arch: str):
     # print(f"amdgcn_bitcode_paths: {amdgcn_bitcode_paths}")
     return amdgcn_bitcode_paths
 
+def get_arch_name() -> str:
+    arch_info = _triton.get_arch_info()
+    gfx_arch_details = re.search('amd.*', arch_info).group(0).strip().split('--')
+    arch_name_features = gfx_arch_details[1].split(':')
+    return arch_name_features[0]
+
+
+def get_warp_size(gpu_name: str) -> int:
```

Comment:
You don't need to guess the `warp_size` from the arch info string. Triton has a `get_warp_size` function defined in C side.
https://github.com/ROCm/triton/blob/e9ab8a27197f03b947ff2561f5fcb88d8cc88dbb/python/src/triton.cc#L1997-L2002
https://github.com/ROCm/triton/blob/e9ab8a27197f03b947ff2561f5fcb88d8cc88dbb/include/triton/Tools/Sys/GetPlatform.hpp#L251-L292

Reply:
@joviliast, could you please take a look at this comments?